src/CMakeLists.txt

```diff
@@ -12,6 +12,27 @@ set_target_properties(${PROJECT_NAME} PROPERTIES
   VERSION ${FCL_VERSION}
   SOVERSION ${FCL_ABI_VERSION})
 
+# Hide symbols by default
+#################################################
+# Macro to check for visibility capability in compiler
+# Original idea from: https://gitorious.org/ferric-cmake-stuff/
+macro (check_gcc_visibility)
```

Comment:
This also works for clang (and maybe other compilers), so maybe rename to check_compiler_visibility?